File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode dummy= ListNode();
        dummy.next=head;
        ListNode * left=&dummy;
        ListNode * right=head;
        while (right && n>0){
            right=right->next;
            n-=1;
        }
        while(left && right){
            left=left->next;
            right=right->next;
        }
        left->next=left->next->next;
        return dummy.next;
    }
};

File: Data Structures & Algorithms/reorder-linked-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        #for reordering the list u have to split the linked list and then merge it back again
        slow=head
        fast=head.next
        while fast and fast.next:
            slow=slow.next
            fast=fast.next.next
        sec=slow.next
        slow.next=None
        prev=tmp=None
        while sec:
            tmp=sec.next
            sec.next=prev
            prev=sec
            sec=tmp
        sec=prev
        f=head
        while sec:
            tmp1=f.next
            tmp2=sec.next
            f.next=sec
            sec.next=tmp1
            f=tmp1
            sec=tmp2
